Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceSerializer.cpp

```diff
@@ -81,6 +81,13 @@ namespace AzToolsFramework
                 result.Combine(resultInstances);
             }
 
+            {
+                AZ::ScopedContextPath subPathSource(context, "m_linkId");
+
+                result = ContinueStoringToJsonObjectField(
+                    outputValue, "LinkId", &(instance->m_linkId), &InvalidLinkId, azrtti_typeid<AZ::u64>(), context);
+            }
```

Comment:
Would this add the linkId to all doms? I know we remove it before saving, could we move that logic that removes it to SanitizeSavingTemplate maybe?

@srikappa-amzn

Reply:
I did some basic sanity testing with @AMZN-mnaumov and we saw that the linkId is not getting written to disk while saving prefabs and levels. But we can add an additional check in SanitizeSavingTemplate to ensure that the top level template doesn't have linkId in it. That should catch any linkIds that may have crept in.
@AMZN-mnaumov , can you add the below check in SanitizeSavingTemplate right before you copy the storedTemplateDom?
```          
            auto linkIdIterator = storedPrefabDom.FindMember(PrefabDomUtils::LinkIdName);
            if (linkIdIterator != storedPrefabDom.MemberEnd())
            {
                instanceDom.RemoveMember(PrefabDomUtils::LinkIdName);
            }

Reply:
Does this need to be added through the Json Serialization? Feels rather hacky to have the serializer inject what looks to be data that's meant for runtime and then depend on a feature (default stripping in this case) to not store it in the final document. Since it's an instance, which I think is on a known path in the PrefabDOM, can't the link id not be added manually outside the Json Serialization?

Reply:
Potentially, yes. That's what we have been doing so far. The problem is that with nested instances being handled by the serializer, there's no single place that can handle restoring the linkId for every case and it becomes quite messy to ensure it is always valid.

To me, the best solutions would either be this (mind you that we already store different doms for internal usage at runtime and storage to file anyways) or simply make GetLinkId() a lazy function that retrieves that value on the spot.

Reply:
Perhaps it's then better to enable the storing of the LinkId explicitly through the meta data from the settings. Something simple like an empty struct that acts as a tag should be enough. If it in the meta data store the LinkId otherwise don't. Doing it that way at least turns it off explicitly instead of it just being a side effect of the default stripping. Also makes it easier to track this case down again down the road once/if a better approach to iterating over Prefabs is build.